src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -186,6 +188,21 @@ private function createRelationsForItem($item, $formattedData)
                     $modelInstance = new $model($relationData['values']);
                     $relation->save($modelInstance);
                 }
+            } elseif ($relation instanceof HasMany) {
+                $modelInstance = $relation->getRelated();
+
+                if ($relationData['values'][$relationMethod] !== null) {
+                    $modelInstance->whereIn($modelInstance->getKeyName(), $relationData['values'][$relationMethod])
+                            ->update([$relation->getForeignKeyName() => $item->{$relation->getLocalKeyName()}]);
+                }else{
+                    //if the foreign key is not nullable we delete the record from the table.
+                    if(!$modelInstance->isColumnNullable($relation->getForeignKeyName())) {
+                        $modelInstance->where($relation->getForeignKeyName(), $item->{$relation->getLocalKeyName()})->delete();
+                    }else{
+                        $modelInstance->where($relation->getForeignKeyName(), $item->{$relation->getLocalKeyName()})
+                                ->update([$relation->getForeignKeyName() => null]);
+                    }
+                }
```

Comment:
Hi @pxpm!

The problem is not completely solved with this. When I delete an existing child and click save, the array of "remaining" children is received correctly here: $relationData['values'][$relationMethod]. So execution enters the if section, where the list of children is correctly updated, but the deleted one is not touched, because it is not in the array.

So besides the update statement, either you get somehow the deleted ids (in case there are more than one deleted) and delete them, or the same if/else statement checking isColumnNullable to delete/set null should be executed.

I hope what I say makes sense to you :smile: Sorry, I don't know the insides so deep to advise about what's the best way to approach this.

Reply:
I got it. Will fix the PR soon. 

Thank you very much for time spent testing and reviewing this, it's very appreciated. 

Best,
Pedro